src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/custom.py

```diff
@@ -0,0 +1,260 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrestazure.tools import is_valid_resource_id, resource_id
+from azure.cli.core.commands.client_factory import get_subscription_id
+
+
+def list_policy_events(
+        cmd,
+        client,
+        management_group_name=None,
+        resource_group_name=None,
+        resource=None,
+        resource_provider_namespace=None,
+        resource_type=None,
+        policy_set_definition_name=None,
+        policy_definition_name=None,
+        policy_assignment_name=None,
+        from_value=None,
+        to_value=None,
+        order_by_clause=None,
+        select_clause=None,
+        top_value=None,
+        filter_clause=None,
+        apply_clause=None):
+
+    from azure.mgmt.policyinsights.models import QueryOptions
+
+    query_options = QueryOptions(
+        top=top_value,
+        order_by=order_by_clause,
+        select=select_clause,
+        from_property=from_value,
+        to=to_value,
+        filter=filter_clause,
+        apply=apply_clause)
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if policy_assignment_name is not None:
+        if resource_group_name is not None:
+            events = client.list_query_results_for_resource_group_level_policy_assignment(
+                subscription_id,
+                resource_group_name,
+                policy_assignment_name,
+                query_options)
+        else:
+            events = client.list_query_results_for_subscription_level_policy_assignment(
+                subscription_id,
+                policy_assignment_name,
+                query_options)
+    elif policy_definition_name is not None:
+        events = client.list_query_results_for_policy_definition(
+            subscription_id,
+            policy_definition_name,
+            query_options)
+    elif policy_set_definition_name is not None:
+        events = client.list_query_results_for_policy_set_definition(
+            subscription_id,
+            policy_set_definition_name,
+            query_options)
+    elif resource is not None:
+        if not is_valid_resource_id(resource):
+            resource = resource_id(
+                subscription=subscription_id,
+                resource_group=resource_group_name,
+                namespace=resource_provider_namespace,
+                type=resource_type,
+                name=resource)
+        events = client.list_query_results_for_resource(
+            resource,
+            query_options)
```

Comment:
Please provide a test for resource id scenario.
Also, what happens with more complex resource-ids like subnets? Here's an example id:
`/subscriptions/<sub id>/resourceGroups/clitestlab/providers/Microsoft.Network/virtualNetworks/Dtlcliautomationlab/subnets/DtlcliautomationlabSubnet`

Reply:
As discussed, I will add a test for that with a hard coded resource id which will not be enabled for live execution.